src/main/java/com/thebois/views/TextureUtils.java

```diff
@@ -32,19 +36,85 @@ public static Texture createCircleTexture(final int radius) {
     }
 
     /**
-     * Creates a new instance of a square shaped texture with the given sideLength length.
+     * Creates a new instance of a circle shaped texture with the given radius.
+     *
+     * @param radius The radius of the circle texture, in pixels.
+     * @param color  The color used in the texture.
+     *
+     * @return The circle texture.
+     */
+    public static Texture createCircleTexture(final int radius, final Color color) {
+        final Pixmap pixmap = new Pixmap(radius * 2, radius * 2, Pixmap.Format.RGBA8888);
+        pixmap.setColor(color);
+        pixmap.fillCircle(radius, radius, radius);
+        final Texture circleTexture = new Texture(pixmap, true);
+        circleTexture.setFilter(Texture.TextureFilter.MipMapLinearLinear,
+                                Texture.TextureFilter.MipMapLinearLinear);
+        pixmap.dispose();
+        return circleTexture;
+    }
+
+    /**
+     * Creates a new instance of a square shaped texture with the side length.
+     * <p>
+     * Sets the base color to white.
+     * </p>
      *
-     * @param sideLength The radius of the circle texture, in pixels.
+     * @param sideLength The size of the square texture, in pixels.
      *
      * @return The square texture.
      */
     public static Texture createSquareTexture(final float sideLength) {
-        final int roundedSideLength = Math.round(sideLength);
-        final Pixmap pixmap = new Pixmap(roundedSideLength,
-                                         roundedSideLength,
-                                         Pixmap.Format.RGBA8888);
-        pixmap.setColor(Color.WHITE);
-        pixmap.fillRectangle(0, 0, roundedSideLength, roundedSideLength);
+        return createSquareTexture(sideLength, Color.WHITE);
+    }
+
+    /**
+     * Creates a new instance of a square shaped texture with the side length.
+     *
+     * @param sideLength The size of the square texture, in pixels.
+     * @param color      The color used in the texture.
+     *
+     * @return The square texture.
+     */
+    public static Texture createSquareTexture(final float sideLength, final Color color) {
+        final int roundedTileSize = (int) sideLength;
```

Comment:
Question

Guessing we had to do this? Feels like this can lead to textures being a different size than the actual tile?

Reply:
Like a house not covering the whole tile

Reply:
Isn't this already in pixels? No such thing as half a pixel, so why the float?

Reply:
It's a float before since our tileSize is a float, and the reason we convert it to int later is because the Pixmap needs an int.
Either we cast our tileSize before you call the createTexture method or the textureMethod does it for you, which one do we want?

Reply:
The method is creating a texture, and textures are in pixels. Pixels can not be specified in floating point units, so the method parameter should be an int. If a cast needs to be done, that is not the responsibility of this method, imo.